lib/pipelineTemplateVersionFactory.js

```diff
@@ -0,0 +1,206 @@
+'use strict';
+
+const schema = require('screwdriver-data-schema');
+const BaseFactory = require('./baseFactory');
+const pipelineTemplateVersions = require('./pipelineTemplateVersions');
+const TEMPLATE_NAME_REGEX = schema.config.regex.FULL_TEMPLATE_NAME;
+const EXACT_VERSION_REGEX = schema.config.regex.EXACT_VERSION;
+const VERSION_REGEX = schema.config.regex.VERSION;
+const TAG_REGEX = schema.config.regex.TEMPLATE_TAG_NAME;
+let instance;
+
+class pipelineTemplateVersionFactory extends BaseFactory {
+    /**
+     * Construct a TemplateFactory object
+     * @method constructor
+     * @param  {Object}     config
+     * @param  {Datastore}  config.datastore         Object that will perform operations on the datastore
+     */
+    constructor(config) {
+        super('pipelineTemplateVersions', config);
+    }
+
+    /**
+     * Instantiate a Template class
+     * @method createClass
+     * @param  {Object}     config               Template data
+     * @param  {Datastore}  config.datastore     Object that will perform operations on the datastore
+     * @param  {String}     config.name          The template name
+     * @param  {String}     [config.namespace]   The template namespace
```

Comment:
`namespace` is not optional
```suggestion
     * @param  {String}     config.namespace   The template namespace
```